packages/ui/origin-ui-core/src/components/certificates/CertificatesInbox.tsx

```diff
@@ -0,0 +1,469 @@
+
+    const updateView = () => {
+        const newViewData = producingDevices
+            .map((device) => {
+                const deviceId = getDeviceId(device, environment);
+                const certs = certificates.filter(
+                    (c) => c.assetId !== undefined && c.deviceId === deviceId
+                );
+
+                return { certs, device };
+            })
+            .filter((d) => d.certs.length > 0)
+            .map((item) => {
+                return {
+                    id: item.device.id.toString(),
+                    name: item.device.facilityName,
+                    country: item.device.country,
+                    type: item.device.deviceType,
+                    capacity: item.device.capacityInW,
+
+                    certificates: item.certs.map((cert) => {
+                        const en = cert.energy.publicVolume.toNumber() / 1000000;
+                        return {
+                            id: cert.id.toString(),
+                            dateStart: cert.generationStartTime,
+                            dateEnd: cert.generationEndTime,
+                            energy: en,
+                            maxEnergy: en,
+                            source: cert.source,
+                            assetId: cert.assetId
+                        };
+                    })
+                };
+            });
+
+        setViewData(newViewData);
+        setPrice(0);
+    };
+
+    useEffect(() => {
+        console.log('certificates updated');
+        updateView();
+    }, [certificates]);
+
+    function getSelectedCertificates(): IInboxCertificateData[] {
+        const cs: Record<string, IInboxCertificateData> = {};
+        viewData.forEach((d) => {
+            d.certificates.forEach((c) => {
+                cs[c.id] = c;
+            });
+        });
+
+        const certs: IInboxCertificateData[] = [];
+
+        selectedCerts.map((cId) => {
+            if (cs[cId]) {
+                certs.push(cs[cId]);
+            }
+        });
+
+        return certs;
+    }
+
+    function updateTotalVolume() {
+        let total = 0;
+
+        getSelectedCertificates().forEach((c) => {
+            total += c.energy;
+        });
+
+        setTotalVolume(total);
+    }
+
+    useEffect(() => {
+        updateTotalVolume();
+    }, [selectedCerts]);
+
+    // toggle on/off selection on all certificates
+    const selectAll = () => {
+        if (allSelected) {
+            setAllSelected(false);
+            setSelectedDevices([]);
+            setSelectedCerts([]);
+        } else {
+            setAllSelected(true);
+            setSelectedDevices(producingDevices.map((d) => d.id.toString()));
+            setSelectedCerts(certificates.map((c) => c.id.toString()));
+        }
+    };
+
+    const areAllDeviceCertSelected = (
+        device: IInboxItemData,
+        selectedCertsList: string[]
+    ): boolean => {
```

Comment:
This could simply be a variable.
No need in creating a function with a loop inside to check it.

Reply:
i don't think this would work.
after clicking checkbox of certificate it should also change checkbox of whole device, to do this i need to check if every certificate of given device is checked/unchecked to update it